q2_types/per_sample_sequences/_transformer.py

```diff
@@ -40,39 +66,29 @@ def _1(dirfmt: SingleLanePerSampleSingleEndFastqDirFmt) \
     result = PerSampleDNAIterators()
     # ensure that dirfmt stays in scope as long as result does
     result.__dirfmt = dirfmt
-    fh = iter(dirfmt.manifest.view(FastqManifestFormat).open())
-    next(fh)
-    for line in fh:
-        sample_id, filename, _ = line.split(',')
-        filepath = str(dirfmt.path / filename)
-        result[sample_id] = skbio.io.read(filepath, format='fastq',
-                                          constructor=skbio.DNA)
+
+    manifest = _prepare_manifest(dirfmt)
+
+    for sample_id, data in manifest.data.iteritems():
+        result[sample_id] = data
```

Comment:
Good point, although I guess you could just pass the dict to the subclass. `dicts` consume `dicts` in their constructor for some reason.